src/sidebar/components/tag-editor.js

```diff
@@ -1,35 +1,197 @@
+    // Call filter with an empty string to return all suggestions
+    return removeDuplicates(tagsService.filter(''), tagList);
+  }, [tagsService, tagList]);
+
+  /**
+   * Handle changes to this annotation's tags
+   */
+  const updateTags = tagList => {
+    // update suggested tags list via service
+    tagsService.store(tagList.map(tag => ({ text: tag })));
+    // invoke callback with updated tags
+    onEditTags({ tags: tagList });
   };
 
-  this.autocomplete = function(query) {
-    return Promise.resolve(tags.filter(query));
+  /**
+   * Remove a tag from this annotation.
+   *
+   * @param {string} tag
+   */
+  const removeTag = tag => {
+    const newTagList = [...tagList]; // make a copy
+    const index = newTagList.indexOf(tag);
+    newTagList.splice(index, 1);
+    updateTags(newTagList);
   };
 
-  this.$onChanges = function(changes) {
-    if (changes.tags) {
-      this.tagList = changes.tags.currentValue.map(function(tag) {
-        return { text: tag };
-      });
+  /**
+   * Adds a tag to the annotation equal to the value of the input field
+   * and then clears out the suggestions list and the input field.
+   */
+  const addTag = () => {
+    const value = inputEl.current.value.trim();
+    if (value.length === 0) {
+      // don't add an empty tag
+      return;
+    }
+    if (tagList.indexOf(value) >= 0) {
+      // don't add duplicate tag
+      return;
+    }
+    updateTags([...tagList, value]);
+    setShowSuggestions(false);
+
+    // clear the input field and maintain focus
+    inputEl.current.value = '';
+    inputEl.current.focus();
+  };
+
+  /**
+   *  If the user pressed enter or comma while focused on
+   *  the <input>, then add a new tag.
+   */
+  const handleKeyPress = e => {
+    if (e.key === 'Enter' || e.key === ',') {
+      addTag();
+      // preventDefault stops the delimiter from being
+      // added to the input field
+      e.preventDefault();
     }
   };
+
+  const handleOnInput = e => {
+    if (e.inputType === 'insertText') {
+      // Show the suggestions if the user types something into the field
+      setShowSuggestions(true);
+    } else if (
+      e.inputType === undefined ||
+      e.inputType === 'insertReplacementText'
+    ) {
+      // nb. Chrome / Safari reports undefined and Firefox reports 'insertReplacementText'
+      // for the inputTyp value when clicking on an element in the datalist.
+      //
+      // There are two ways to arrive here, either click an item with the mouse
+      // or use keyboard navigation and press 'Enter'.
+      //
+      // If the input value typed already exactly matches the option selected
+      // then this Event won't fire and a user would have to press 'Enter' a second
+      // time to trigger the handleKeyPress callback above to add the tag.
+      addTag();
+    } else if (inputEl.current.value.length === 0) {
+      // If the user deleted input, hide suggestions. This has
+      // no effect in Safari and the list will stay open.
```

Comment:
Can you add a link to the GitHub issue where we are tracking known issues with Safari. In general, if you find a need to either work around a browser bug or just call attention to the fact that it exists, it is very helpful if the code includes a link to a relevant bug report so that future readers can check whether the workaround is still relevant or whether they need to be aware of it when making future changes.